recipes/Switchboard/ASR/CTC/train_with_wav2vec.py

```diff
@@ -0,0 +1,438 @@
+            csv_path=csv_file, replacements={"data_root": data_folder}
+        )
+        test_datasets[name] = test_datasets[name].filtered_sorted(
+            sort_key="duration"
+        )
+    datasets = [train_data, valid_data] + [i for _, i in test_datasets.items()]
+
+    # 2. Define audio pipeline:
+    @sb.utils.data_pipeline.takes("wav", "channel", "start", "stop")
+    @sb.utils.data_pipeline.provides("sig")
+    def audio_pipeline(wav, channel, start, stop):
+        # Select a speech segment from the sph file
+        # start and end times are already frames.
+        # This is done in data preparation stage.
+        start = int(start)
+        stop = int(stop)
+        num_frames = stop - start
+        sig, fs = torchaudio.load(
+            wav, num_frames=num_frames, frame_offset=start
+        )
+        info = torchaudio.info(wav)
+
+        resampled = sig
+        # Maybe resample to 16kHz
+        if int(info.sample_rate) != int(hparams["sample_rate"]):
+            resampled = torchaudio.transforms.Resample(
+                info.sample_rate, hparams["sample_rate"],
+            )(sig)
+
+        resampled = resampled.transpose(0, 1).squeeze(1)
+        # Select the proper audio channel of the segment
+        if channel == "A":
+            resampled = resampled[:, 0]
+        else:
+            resampled = resampled[:, 1]
+        return resampled
+
+    sb.dataio.dataset.add_dynamic_item(datasets, audio_pipeline)
+
+    # 3. Define text pipeline:
+    @sb.utils.data_pipeline.takes("words")
+    @sb.utils.data_pipeline.provides(
+        "tokens_list", "tokens_bos", "tokens_eos", "tokens"
+    )
+    def text_pipeline(wrd):
+        tokens_list = tokenizer.sp.encode_as_ids(wrd)
+        yield tokens_list
+        tokens_bos = torch.LongTensor([hparams["bos_index"]] + (tokens_list))
+        yield tokens_bos
+        tokens_eos = torch.LongTensor(tokens_list + [hparams["eos_index"]])
+        yield tokens_eos
+        tokens = torch.LongTensor(tokens_list)
+        yield tokens
+
+    sb.dataio.dataset.add_dynamic_item(datasets, text_pipeline)
+
+    # 4. Set output:
+    sb.dataio.dataset.set_output_keys(
+        datasets, ["id", "sig", "tokens_bos", "tokens_eos", "tokens"],
+    )
+    return train_data, valid_data, test_datasets
+
+
+if __name__ == "__main__":
+
+    # Load hyperparameters file with command-line overrides
+    hparams_file, run_opts, overrides = sb.parse_arguments(sys.argv[1:])
+    with open(hparams_file) as fin:
+        hparams = load_hyperpyyaml(fin, overrides)
+
+    # If distributed_launch=True then
+    # create ddp_group with the right communication protocol
+    sb.utils.distributed.ddp_init_group(run_opts)
+
+    # Dataset preparation (parsing Switchboard)
+    from switchboard_prepare import prepare_switchboard  # noqa
+    from normalize_util import normalize_words, read_glm_csv  # noqa
+
+    normalize_fn = functools.partial(
```

Comment:
This needs to be called AFTER `prepare_switchboard` is executed or you'll get a `FileNotFoundError` because the glm file won't be available yet.

Reply:
Fixed in 34c0f62aa49b08155766997b443de58c13dcf399.